src/tty/unix.rs

```diff
@@ -378,6 +400,40 @@ impl EventedReadWrite for Pty {
     }
 }
 
+impl EventedPty for Pty {
+    #[inline]
+    fn get_child_event(&mut self) -> Option<ChildEvent> {
+        let sig = {
+            let mut sig = [0u8];
+            self.sig_pipe_read.read_exact(&mut sig).unwrap();
+            i32::from(sig[0])
+        };
+
+        if sig == libc::SIGCHLD {
+            match self.child.try_wait() {
+                Err(e) => {
+                    error!("Error checking child process termination: {}", e);
+                    None
+                },
+                Ok(None) => None,
+                Ok(_)    => Some(ChildEvent::Exited),
+            }
+        }
+        else {
+            None
+        }
+    }
+
+    #[inline]
+    fn child_event_token(&self) -> mio::Token {
+        self.sig_token
+    }
+}
+
+pub fn process_should_exit() -> bool {
+    false // TODO: remove this stub if/when windows no longer needs it
```

Comment:
By this I'd assume you mean that we should implement the `get_child_event` method on the winpty and return a new child event when the PTY should quit on Windows?

If this is the case and you don't want to include it in this PR (which is perfectly fine by me, since that's kinda separate anyways), I'd propose opening an issue and removing this TODO, so it's tracked in github. Alternatively the TODO could also document what actually needs to be done and maybe put a link to the issue.